pkg/util/deployment/deployment.go

```diff
@@ -40,6 +40,12 @@ import (
 const (
 	// The revision annotation of a deployment's replica sets which records its rollout sequence
 	RevisionAnnotation = "deployment.kubernetes.io/revision"
+	// DesiredReplicasAnnotation is the desired replicas for a deployment recorded as an annotation
+	// in its replica sets.
+	DesiredReplicasAnnotation = "deployment.kubernetes.io/desired-replicas"
+	// TotalReplicasAnnotation is the total replicas for a deployment (desired+maxSurge in case the
+	// new replica set is not satured) recorded in its replica sets.
+	TotalReplicasAnnotation = "deployment.kubernetes.io/total-replicas"
```

Comment:
This is not actual replicas, it is the total replicas that our rolling deployment can accomodate:
* d.Spec.Replicas in case the new replica set is saturated (no rollout ongoing)
* d.Spec.Replicas+d.Spec.Rolling.MaxSurge in case there is a rollout ongoing

It is useful when we estimate proportions for different replica sets.

 during a rollout (which is d.Spec.Replicas+d.Spec.Rolling.MaxSurge in case there are multiple